janus_server/src/bin/provision_task.rs

```diff
@@ -0,0 +1,203 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use janus_core::time::{Clock, RealClock};
+use janus_server::{
+    binary_utils::{janus_main, BinaryContext, BinaryOptions, CommonBinaryOptions},
+    config::{BinaryConfig, CommonConfig},
+    task::Task,
+};
+use serde::{Deserialize, Serialize};
+use structopt::StructOpt;
+use tracing::info;
+
+static SCHEMA: &str = include_str!("../../../db/schema.sql");
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    janus_main::<Options, _, Config, _, _>(RealClock::default(), run).await
+}
+
+async fn run<C: Clock>(ctx: BinaryContext<C, Config>) -> Result<()> {
+    // Try to write the DB schema, if requested.
+    if ctx.config.write_schema {
+        info!("Writing database schema");
+        let db_client = ctx.pool.get().await?;
+        db_client.batch_execute(SCHEMA).await?;
+    }
+
+    // Write all tasks requested.
+    if !ctx.config.tasks.is_empty() {
+        let tasks = Arc::new(ctx.config.tasks);
+        info!(task_count = tasks.len(), "Writing tasks");
+        ctx.datastore
+            .run_tx(|tx| {
+                let tasks = Arc::clone(&tasks);
+                Box::pin(async move {
+                    for task in tasks.iter() {
+                        tx.delete_task(task.id).await?;
```

Comment:
I think this delete-then-put behavior could lead to confusing emergent behavior. For example, if this command is run with multiple tasks in its configuration, but one task already exists in the database and has a client report as well, then this whole transaction will roll back, and none of the other tasks will be created.

My preference would be that `delete_task()` should use `check_update()` on the number of rows deleted from the tasks table, and provision_task should just create tasks without deleting them first, perhaps skipping any task IDs that are already present.